src/transformers/generation/logits_process.py

```diff
@@ -1102,3 +1102,111 @@ def __call__(self, input_ids, scores):
+
+    Args:
+        guidance_scale (float):
+            The guidance scale for classifier free guidance (CFG). CFG is enabled by setting `guidance_scale > 1`.
+            Higher guidance scale encourages the model to generate samples that are more closely linked to the input
+            prompt, usually at the expense of poorer quality.
+        unconditional_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Indices of input sequence tokens in the vocabulary for the unconditional branch. If unset, will default to
+            the last token of the prompt.
+        model:
+            The LM computing the unconditional scores. Supposedly the same as the one computing the conditional scores.
+            Both models must use the same tokenizer.
+        use_cache (bool, **optional**):
+            Whether to cache key/values during the negative prompt forward pass.
+        unconditional_attention_mask (`torch.LongTensor` of shape `(batch_size, sequence_length)`, **optional**):
+            Attention mask for unconditional_ids.
+
+
+    Examples:
+
+    ```python
+    >>> # base prompt emphasis example
+    >>> from transformers import AutoTokenizer, AutoModelForCausalLM
+
+    >>> model = AutoModelForCausalLM.from_pretrained("gpt2")
+    >>> tokenizer = AutoTokenizer.from_pretrained("gpt2")
+    >>> inputs = tokenizer(["Today, a dragon flew over Paris, France,"], return_tensors="pt")
+    >>> summary_ids = model.generate(inputs["input_ids"], guidance_scale=1.5)
+    >>> print(tokenizer.batch_decode(summary_ids, skip_special_tokens=True)[0])
+    The dragon flew over Paris, France, landing in Lyon, a city of a few million. Dragon-flying was a new form of
+    transport, and the dragon was the first in Europe.
+
+    >>> # with a negative prompt
+    >>> neg_inputs = tokenizer(["A very happy event happened,"], return_tensors="pt")
+    >>> summary_ids = model.generate(inputs["input_ids"], guidance_scale=2, negative_prompt=neg_inputs["input_ids"])
+    >>> print(tokenizer.batch_decode(summary_ids, skip_special_tokens=True)[0])
+    The dragon flew over Paris, France, crashing into Notre Dame Cathedral in the French capital killing at least 127
+    people and injuring more than 350.
+    ```
+    """
+
+    def __init__(
+        self,
+        guidance_scale,
+        unconditional_ids,
+        model,
+        use_cache=True,
+        unconditional_attention_mask=None,
+    ):
+        self.guidance_scale = guidance_scale
+        self.model = model
+        self.negative_context = {
+            "input_ids": unconditional_ids,
+            "attention_mask": unconditional_attention_mask or torch.ones_like(unconditional_ids, dtype=torch.long),
+            "use_cache": use_cache,
+            "past_key_values": None,
+            "first_pass": True,
+        }
+
+    def neg_logits(self, input_ids):
+        ctx = self.negative_context
+
+        if "first_pass" in ctx:
```

Comment:
nit: checking `if ctx["first_pass"] is True:` and then setting it to false with `ctx["first_pass"] = False` would be more in line with how we code in `transformers` :)